connectors/src/connectors/microsoft/lib/content_nodes.ts

```diff
@@ -161,3 +162,59 @@ export function getFolderAsContentNode(
     permission: "none",
   };
 }
+
+export function getFileAsContentNode(
+  file: microsoftgraph.DriveItem,
+  parentInternalId: string
+): ContentNode {
+  return {
+    provider: "microsoft",
+    internalId: internalIdFromTypeAndPath({
```

Comment:
this is refactored on the incremental sync PR by ag getDriveItemInternalId function
I will merge the incr sync pr even if it requires some more testing to avoid diverging with our various refactorings